use-cases/rag-on-gke/alloyDB/init-database-objects/assets/ml-integration.sql

```diff
@@ -0,0 +1,100 @@
+-- set these endpoints from environment variables
+-- To use this file, you can run psql command like this
+-- # export FINETUNE_MODEL_EP=<your-finetuned-model-endpoint>
+-- # export PRETRAINED_MODEL_EP=<your-pretained-model-endpoint>
+-- # export EMBEDDING_ENDPOINT=<your-embedding-service-ebdpoint>
+-- # psql <your-connection-string> -f <this-file>
+\getenv pretrained_model_ep PRETRAINED_MODEL_EP
+\getenv embedding_endpoint EMBEDDING_ENDPOINT
+
+-- If you don't want to use environment variable, uncomment the following lines
+-- \set finetune_model_ep http://10.150.0.32:8000/v1/completions
+-- \set pretrained_model_ep http://10.150.0.23:8000/v1/completions
+-- \set embedding_endpoint http://10.150.15.227/embeddings
+
+-- set google_ml_integration.enable_model_support = 'on';
+
```

Comment:
What do we gain by doing this with store procedures instead of doing this an abstraction layer outside of the database?

Reply:
These are not stored procedures. They are functions. 

Stored procedures are to run combined bussiness logic inside database, which is not encouraged anymore. 

Functions are extension to database, for example, json related operations are done by functions. another example is pgvector, the vector operations are also function. 

The objects created in this file are all functions. this is following the practise of "google_ml_integration" extension of AlloyDB(and CloudSQL postgresql)